java/java.project.ui/src/org/netbeans/modules/java/project/ui/NewJavaFileWizardIterator.java

```diff
@@ -310,7 +314,12 @@ private void addRequires(FileObject createdFile, Set<String> requiredModuleNames
             createdFile = dobj.getPrimaryFile();
         } else {
             DataObject dTemplate = DataObject.find( template );                
-            DataObject dobj = dTemplate.createFromTemplate( df, targetName );
+            String superclass = (String) wiz.getProperty(SUPERCLASS);
+            String interfaces = (String) wiz.getProperty(INTERFACES);
+            Map<String, String> parameters = new HashMap<>();
+            parameters.put(SUPERCLASS, superclass);
```

Comment:
This is a change of API and should be documented as such (incremement module version, enhance `arch.xml`, note in `apichanges.xml`).